Dockerfile

```diff
@@ -3,6 +3,10 @@
 # https://github.com/docker-library/repo-info/tree/master/repos/alpine/remote
 FROM alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad
 ARG EXTENSION_FILE
+ARG BUILD_DATE
```

Comment:
**issue**: I don't think this is right. The dockerfile is only copying the build artifacts to the docker image. There should be no need to change the file time inside the docker image as the build artifacts should have already been built reproducibly.

Reply:
The reason for this way of doing it is a "limitation" of GoReleaser as it copies the artifacts and files used in the Dockerfile to a temporary directory and builds it from there.

You can see this behavior when running `make dist` 

<img width="421" alt="image" src="https://user-images.githubusercontent.com/16325797/215425026-9d5c551f-0714-4cbe-a7dc-8332ba57223d.png">

Reply:
I tried running the command and got the following:

```
release failed after 1s                  error=docker build failed: failed to build docker.elastic.co/observability/apm-lambda-extension-arm64:1.2.0: exit status 125: unknown flag: --builder
```

it seems goreleaser is using `docker-buildx` under the hood. Installing `docker-buildx` fixed the issues. (should be documented, opened a followup issue: https://github.com/elastic/apm-aws-lambda/issues/372)

building the docker image is failing but I think that's related to a typo. I've left a comment below.

Reply:
Ok, I see some progress but the build is still failing:

```
#8 [4/4] RUN find /opt -exec touch -amdt "1675338235" {} ;
#8 0.240 exec /bin/sh: exec format error
#8 ERROR: executor failed running [/bin/sh -c find /opt -exec touch -amdt "${COMMIT_TIMESTAMP}" {} \;]: exit code: 1
------
 > [4/4] RUN find /opt -exec touch -amdt "1675338235" {} ;:
#8 0.240 exec /bin/sh: exec format error
------
ERROR: failed to solve: executor failed running [/bin/sh -c find /opt -exec touch -amdt "${COMMIT_TIMESTAMP}" {} \;]: exit code: 1
```

Reply:
@reakaleek I think this is the last issue. Is the docker build working for you ?

Reply:
Interesting. Yes, it works for me. (Apple macOS M1)

And this [workflow](https://github.com/elastic/apm-aws-lambda/actions/runs/4076401474/jobs/7024085307) shows that it should work with Ubuntu on amd64.

Looking at your error message, if I recall correctly, this could be a buildx setup issue.

Reply:
I was able to reproduce the issue by building the arm64 image.

`docker buildx build --platform linux/arm64 .`

I'm on linux (amd64). I'll investigate more